src/Sarif.Multitool.Library/Rules/SARIF2016.FileUrisShouldBeRelative.cs

```diff
@@ -72,14 +73,32 @@ private void AnalyzeResultLocations(IList<Location> locations, string locationsP
 
                 if (location?.PhysicalLocation?.ArtifactLocation?.Uri != null)
                 {
-                    if (location.PhysicalLocation.ArtifactLocation.Uri.IsAbsoluteUri && location.PhysicalLocation.ArtifactLocation.Uri.IsFile)
+                    if (location.PhysicalLocation.ArtifactLocation.Uri.IsAbsoluteUri)
                     {
-                        // {0}: The file location '{1}' is specified with absolute URI. Prefer a relative
-                        // reference together with a uriBaseId property.
-                        LogResult(
-                            locationPointer.AtProperty(SarifPropertyName.PhysicalLocation).AtProperty(SarifPropertyName.ArtifactLocation).AtProperty(SarifPropertyName.Uri),
-                            nameof(RuleResources.SARIF2016_FileUrisShouldBeRelative_Note_Default_Text),
-                            location.PhysicalLocation.ArtifactLocation.Uri.OriginalString);
+                        if (location.PhysicalLocation.ArtifactLocation.Uri.IsFile)
+                        {
+                            // {0}: The file location '{1}' is specified with absolute URI. Prefer a relative
+                            // reference together with a uriBaseId property.
+                            LogResult(
+                                locationPointer.AtProperty(SarifPropertyName.PhysicalLocation).AtProperty(SarifPropertyName.ArtifactLocation).AtProperty(SarifPropertyName.Uri),
+                                nameof(RuleResources.SARIF2016_FileUrisShouldBeRelative_Note_Default_Text),
+                                location.PhysicalLocation.ArtifactLocation.Uri.OriginalString);
+                        }
+                    }
+                    else
+                    {
+                        if (location.PhysicalLocation.ArtifactLocation.Uri.OriginalString.StartsWith("/"))
+                        {
+                            // {0}: The file location '{1}' start with slash. Prefer a relative reference
+                            // together with a uriBaseId property.
```

Comment:
let's use this text:

{0}: The relative file URL '{1}' is prefixed with a leading slash, which can lead to unintended behavior when concatenating with absolute URLs. Remove the leading slash.